apps/laboratory/src/hooks/useWagmiActiveCapabilities.ts

```diff
@@ -50,25 +53,37 @@ export function useWagmiAvailableCapabilities({
       chainId: connectedChain.id
     })
     if (connectedProvider instanceof EthereumProvider) {
-      setEthereumProvider(connectedProvider)
-      let walletCapabilities = undefined
-      walletCapabilities = getProviderCachedCapabilities(connectedAccount, connectedProvider)
+      setProvider(connectedProvider)
+      const walletCapabilities = getProviderCachedCapabilities(connectedAccount, connectedProvider)
+      setAvailableCapabilities(walletCapabilities)
+    } else if (connectedProvider instanceof W3mFrameProvider) {
+      const walletCapabilities = await connectedProvider.getCapabilities()
+      setProvider(connectedProvider)
       setAvailableCapabilities(walletCapabilities)
     }
   }
 
   function isMethodSupported(): boolean {
-    return Boolean(
-      ethereumProvider?.signer?.session?.namespaces?.['eip155']?.methods?.includes(method)
-    )
+    if (provider instanceof W3mFrameProvider) {
+      return ['wallet_sendCalls', 'wallet_getCapabilities', 'wallet_getCallsStatus'].includes(
```

Comment:
why this hardcoded? should W3mFrameProvider expose its capabilities?

Reply:
maybe is the provider.getCapabilities()

Reply:
Not the same. Capabilities !== rpc methods supported